src/community/graticule/src/main/java/org/geotools/data/graticule/GraticuleDataStore.java

```diff
@@ -0,0 +1,388 @@
+import org.geotools.api.data.Transaction;
+import org.geotools.api.feature.simple.SimpleFeature;
+import org.geotools.api.feature.simple.SimpleFeatureType;
+import org.geotools.api.feature.type.Name;
+import org.geotools.api.filter.Filter;
+import org.geotools.api.referencing.crs.CoordinateReferenceSystem;
+import org.geotools.data.CollectionFeatureReader;
+import org.geotools.data.DefaultServiceInfo;
+import org.geotools.data.graticule.gridsupport.LineFeatureBuilder;
+import org.geotools.feature.FeatureTypes;
+import org.geotools.feature.NameImpl;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.grid.Lines;
+import org.geotools.grid.ortholine.LineOrientation;
+import org.geotools.grid.ortholine.OrthoLineDef;
+import org.locationtech.jts.geom.LineString;
+
+public class GraticuleDataStore implements DataStore {
+    static final Logger log = Logger.getLogger("GraticuleDataStore");
+    private final ReferencedEnvelope bounds;
+    private final List<Double> steps;
+    private final HashMap<Name, SimpleFeatureSource> sources = new HashMap<>();
+
+    private final ArrayList<Name> names = new ArrayList<>();
+
+    public GraticuleDataStore(ReferencedEnvelope env, List<Double> steps) {
+        this.steps = steps;
+        this.bounds = env;
+
+        int level = 0;
+        Collections.sort(steps);
+        for (double step : steps) {
+            Name name = new NameImpl(Double.toString(step).replace('.', '_'));
+            SimpleFeatureType schema = buildType(name, bounds.getCoordinateReferenceSystem());
+            log.fine("Creating graticule with name " + name);
+            names.add(name);
+
+            List<OrthoLineDef> lineDefs =
+                    Arrays.asList(
+                            // vertical (longitude) lines
+                            new OrthoLineDef(LineOrientation.VERTICAL, level, step),
+                            // horizontal (latitude) lines
+                            new OrthoLineDef(LineOrientation.HORIZONTAL, level, step));
+
+            // Specify vertex spacing to get "densified" polygons
+            double vertexSpacing = (bounds.getHeight() / 20); // should be dynamic
+            SimpleFeatureSource grid =
+                    Lines.createOrthoLines(
+                            bounds, lineDefs, vertexSpacing, new LineFeatureBuilder(schema));
+            sources.put(name, grid);
+            level++;
+        }
+    }
+
+    private static SimpleFeatureType buildType(Name name, CoordinateReferenceSystem crs) {
+        SimpleFeatureTypeBuilder tb = new SimpleFeatureTypeBuilder();
+        tb.setName(name);
+        tb.add(LineFeatureBuilder.DEFAULT_GEOMETRY_ATTRIBUTE_NAME, LineString.class, crs);
+        tb.add(LineFeatureBuilder.ID_ATTRIBUTE_NAME, Integer.class);
+        tb.add(LineFeatureBuilder.LEVEL_ATTRIBUTE_NAME, Integer.class);
+        tb.add(LineFeatureBuilder.VALUE_LABEL_NAME, String.class);
```

Comment:
I'd avoid having pre-generated labels. Different users might want to display, or not, the unit of measure/direction of the axis, and format large numbers differently. I'd delegate all that to the SLD, eventually adding support functions if needed (but probably all we need is a good example).